kong.conf.default

```diff
@@ -165,6 +173,12 @@
                          # - `transparent` will cause kong to listen to, and
                          #   respond from, any and all IP addresses and ports
                          #   you configure in iptables.
+                         # - `bind` instructs to make a separate bind() call
+                         #   for a given address:port pair.
+                         # - `reuseport` instructs to create an individual
+                         #   listening socket for each worker process
+                         #   allowing a kernel to distribute incoming
+                         #   connections between worker processes
```

Comment:
Same here.

Reply:
Yes, again, should it be enabled by default? Does this concern us: http://man7.org/linux/man-pages/man7/socket.7.html

And yes, previously we did not even give option for these, so in some ways enabling them may lead to unwanted results. Perhaps in most cases to wanted results, though. I leave decision to others.

Reply:
I tested the `reuseport` with our `miniperf` and here are results on my 2010 Mac Pro:
```
without reuseport: 25000 req/s
with reuseport: 8500 req/s
```

So it seems like on this machine the reuseport actually kills the performance. Other setting didn't make measurable difference.

Reply:
@bungle It should be fine to enable it by default. The security implication refers to cases where people runs Nginx using incorrect user permission setup, which in that case is an even bigger security risk thank `SO_REUSEPORT` is.

The way `SO_REUSEPORT` is guarded is that the Kernel requires subsequent socket who tries to listen on the same port to belong to the same user as the first process who opened it. This should be enough to prevent any port hijacking from happening.

Other softwares, such as HAProxy, already enables it by default: https://www.haproxy.org/download/1.8/doc/configuration.txt (search `noreuseport`)

Reply:
@dndx did you see above how it makee the performance suck? I need to retest on Linux.